### DISCUSSION TITLE
Airbyte RabbitMQ Connector Archived (Not Available)

### DISCUSSION THREAD

**Discussion by whalygood** [bionic-gpt/bionic-gpt#407] [General]
In the [Bionic-GPT docs](https://bionic-gpt.com/docs/document-pipeline/airbyte/), it is advised to use the RabbitMQ destination connector to integrate a data source with the Data Pipelines for the custom datasets. However according to the [Airbyte docs](https://docs.airbyte.com/integrations/destinations/rabbitmq), the RabbitMQ connector is "archived" and thereby no longer available in either the OSS or Cloud versions of Airbyte. 

Is there an alternative service for queuing that is recommended for using with the Data Pipelines feature?

**Comment by ianpurton** [comment]
I thought RabbitMQ was available in the OSS version?

I can see it's archived. That's a shame. I'll look for an alternative.

**Comment by ianpurton** [comment]
Looks like rabbit mq was pulled by mistake!!!! https://github.com/airbytehq/airbyte/pull/36749

So I guess it will re-appear at some point.

**Reply by whalygood** [reply]
I redeployed the helm chart for Airbyte and the RabbitMQ connector is back in 0.58.0!
Was able to get a queue and binding set up in RabbitMQ and the test with cURL was successful:
```
curl -i -u admin:admin -H "Content-Type: application/json" -X POST -d '{"properties":{},"routing_key":"123456.bionic-pipeline","payload":"Your_Message_Content","payload_encoding":"string"}' http://localhost:15672/api/exchanges/%2F/amq.topic/publish
HTTP/1.1 200 OK
cache-control: no-cache
content-length: 15
content-security-policy: script-src 'self' 'unsafe-eval' 'unsafe-inline'; object-src 'self'
content-type: application/json
date: Sat, 27 Apr 2024 21:28:55 GMT
server: Cowboy
vary: accept, accept-encoding, origin

{"routed":true}
```

But I encountered this error **gaierror(-3, 'Try again')** when setting up RabbitMQ as a destination in Airbyte.
Looks like something to do with error in name resolution. My service is called rabbitmq-service and my namespace is called rabbitmq though, so I'm not sure what could be wrong with the hostname **rabbitmq-service.rabbitmq.svc.cluster.local**? Have tried with and without the 5672 port - same issue.

![image](https://github.com/bionic-gpt/bionic-gpt/assets/31971061/e84a90ab-bf3e-4239-9274-72bdd8622568)
